src/app/StudentView.tsx

```diff
@@ -63,10 +66,15 @@ const AddGradeOverlay: React.FunctionComponent<{ studentID: number, refreshTrans
                     event.preventDefault();
                     // @ts-ignore
                     const formElements = event.target as HTMLInputElement[];
-                    const courseName = formElements[0].value;
-                    const grade = formElements[1].value;
+                    const courseName = formElements[0].value
+                    let gradeTemp = gradeList.find(val => val === formElements[1].value);
+                    let grade : Grade =  Grade['A'];
+                    if (gradeTemp) {
+                        grade = Grade['A']
+                    }
```

Comment:
Please consider adding a comment here on the use of the variable ‘gradeTemp.’ I find it challenging to understand the purpose of this code section.